sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -265,6 +268,7 @@ case class EnsureRequirements(conf: SQLConf) extends Rule[SparkPlan] {
           if (childPartitioning.guarantees(partitioning)) child else operator
         case _ => operator
       }
-    case operator: SparkPlan => ensureDistributionAndOrdering(operator)
+    case operator: SparkPlan =>
+      ensureDistributionAndOrdering(reorderJoinPredicates.apply(operator))
```

Comment:
Then, do something like 
```Scala
ensureDistributionAndOrdering(ReorderJoinPredicates(operator))
```

Reply:
Could you add a comment to explain why we do it here. It is hard for new comers to understand the assumptions we made here.

Reply:
It feels like having a rule invoked in such fashion is in-consistent compared to rest of the codebase .... from point of view of someone new to codebase, it will look odd. I removed the rule and instead moved this methods inside `EnsureRequirements`. Let me know how you feel about the changed version